tools/federation/src/main/java/org/eclipse/rdf4j/federated/FederationManager.java

```diff
@@ -103,28 +105,46 @@ public void reset() {
 			log.debug("Scheduler for join and union are reset.");
 		}
 
+		Optional<TaskWrapper> taskWrapper = federationContext.getConfig().getTaskWrapper();
 		if (joinScheduler != null) {
 			joinScheduler.abort();
 		}
 		joinScheduler = new ControlledWorkerScheduler<>(federationContext.getConfig().getJoinWorkerThreads(),
 				"Join Scheduler");
+		joinScheduler.setTaskWrapper(taskWrapper);
```

Comment:
You're passing the Optional, rather than the TaskWrapper - will that work?

Reply:
Ah never mind I see what you did - added an overloaded setter. 

I personally prefer handling the Optional at the point of message passing rather than inside the setter:

    federationContext.getConfig().getTaskWrapper().ifPresent(joinScheduler::setTaskWrapper);

but YMMV.

Reply:
That looks better, will change and remove the default implementation for optional. Thanks